src/utils/layerUtils.js

```diff
@@ -97,3 +101,36 @@ export function indexOfMaxResolution(projection, layerMaxResolution) {
     }
     return indexOfResolution
 }
+
+/**
+ * Creates a LayerFeature object from an OpenLayers feature and a layer.
+ *
+ * @param {ol.Feature} olFeature - The OpenLayers feature to convert.
+ * @param {AbstractLayer} layer - The layer associated with the feature.
+ * @returns {LayerFeature | null} The created LayerFeature object or null if the feature has no
+ *   geometry.
+ */
+export function createLayerFeature(olFeature, layer) {
+    if (!olFeature?.getGeometry()) return null
+    return new LayerFeature({
+        layer: layer,
+        id: olFeature.getId(),
+        name:
+            olFeature.get('label') ??
+            // exception for MeteoSchweiz GeoJSONs, we use the station name instead of the ID
+            // some of their layers are
+            // - ch.meteoschweiz.messwerte-niederschlag-10min
+            // - ch.meteoschweiz.messwerte-lufttemperatur-10min
+            olFeature.get('station_name') ??
+            // GPX track feature don't have an ID but have a name !
+            olFeature.get('name') ??
+            olFeature.getId(),
```

Comment:
This code seems to be copy paste from other place(s) ? we should maybe put it into a function and use it everywhere where it is needed